packages/e2e-tests/test/e2e-current-op.spec.ts

```diff
@@ -0,0 +1,84 @@
+import { expect } from 'chai';
+import {
+  skipIfApiStrict,
+  startSharedTestServer,
+} from '../../../testing/integration-testing-hooks';
+import { TestShell } from './test-shell';
+
+describe('e2e currentOp', function () {
+  skipIfApiStrict();
+  afterEach(TestShell.cleanup);
+
+  const testServer = startSharedTestServer();
+
+  context('with 2 shells', function () {
+    let helperShell: TestShell;
+    let currentOpShell: TestShell;
+
+    const OPERATION_TIME = 1_000;
+    const CURRENT_OP_WAIT_TIME = 100;
+    this.timeout(OPERATION_TIME * 2);
+
+    beforeEach(async function () {
+      helperShell = TestShell.start({
+        args: [await testServer.connectionString()],
+      });
+      currentOpShell = TestShell.start({
+        args: [await testServer.connectionString()],
+      });
+      await helperShell.waitForPrompt();
+
+      // Insert a dummy object so find commands will actually run with the delay.
+      await helperShell.executeLine('db.coll.insertOne({})');
+    });
+
+    it('should return the correct operation', async function () {
+      const regexOperation = helperShell.executeLine(
+        `db.coll.find({$where: function() { sleep(${OPERATION_TIME}) }}).projection({test: 1})`
+      );
+      helperShell.assertNoErrors();
+      await currentOpShell.executeLine(`sleep(${CURRENT_OP_WAIT_TIME})`);
+      let currentOpCall = await currentOpShell.executeLine(`db.currentOp()`);
+
+      currentOpShell.assertNoErrors();
+      expect(currentOpCall).to.include("find: 'coll'");
+      expect(currentOpCall).to.include(
+        `filter: { '$where': Code('function() { sleep(${OPERATION_TIME}) }') }`
+      );
+      expect(currentOpCall).to.include('projection: { test: 1 }');
+
+      console.log(currentOpCall);
+
+      await regexOperation;
+
+      currentOpCall = await currentOpShell.executeLine(`db.currentOp()`);
+
+      currentOpShell.assertNoErrors();
+      expect(currentOpCall).not.to.include("find: 'coll'");
+      expect(currentOpCall).not.to.include(
+        `filter: { '$where': Code('function() { sleep(${OPERATION_TIME}) }') }`
+      );
+      expect(currentOpCall).not.to.include('projection: { test: 1 }');
+    });
+
+    it('should work when the operation contains regex', async function () {
+      const regExpString = '^(?i)\\\\Qchho0842\\E';
+      // The values from currentOp removes redundant escapes such as \E
```

Comment:
Seems like expected behavior right?

Reply:
Okay, so, the way I read this, it's not even $currentOp that removes the E, it's just normal JS escaping behavior that applies before we run any code at all; the code here translates to 

```js
executeLine(`db.coll.find(...).projection({re: BSONRegExp('${regExpString}')})`)
```

which expands to 


```js
db.coll.find(...).projection({re: BSONRegExp('^(?i)\\Qchho0842\E')})
```

which is evaluated as JS code, where `'^(?i)\\Qchho0842\E' === '^(?i)\\Qchho0842E'`.

If you want to actually test that $currentOp drops the `\` in front of an `E`, you'd need to escape it like the `Q` here.

I feel like this would be a bit easier to read if we used `String.raw` for defining these literals and `JSON.stringify()` for inserting them into JS code without having to worry about escaping logic; i.e.

```js
const regExpString = String.raw `^(?i)\Qchho0842\E`;
...
helperShell.executeLine(`db.coll.find(...).projection({re: BSONRegExp(${JSON.stringify(regExpString)})})`);
```

Reply:
Okay today was the day I learnt 😄 
```tsx
"\\Q"  === "\\\Q" === String.raw`\Q`
```
Surprised I hadn't used `String.raw` much before, good to know, thank you!